---
layout:     post
title:      "Varscan2"
subtitle:   "call SNV"
date:       2020-07-01 10:00:00
author:     "Meng"
header-img:  "img/20200701-Varscan2.jpg"
catalog: true
tags:
    - 生物信息学相关软件
    - 基因组分析
---


Manual: [Manual](http://varscan.sourceforge.net/using-varscan.html)

1. ##### somatic

   This command calls variants and identifies their somatic status (Germline/LOH/Somatic) using [pileup files](http://varscan.sourceforge.net/using-varscan.html#pileup-howto) from a matched tumor-normal pair.

   ```shell
   USAGE: java -jar VarScan.jar somatic [normal-tumor.mpileup] [output] --mpileup 1 OPTIONS
   normal-tumor.mpileup: The SAMtools mpileup file with normal and then tumor
   output: Output base name for SNP and indel output
   ```

   OPTIONS:

   | Options               |                                                          |
   | --------------------- | -------------------------------------------------------- |
   | --output-snp          | Output file for SNP calls [default: output.snp]          |
   | --output-indel        | Output file for indel calls [default: output.indel]      |
   | --min-coverage        | Minimum coverage in normal and tumor to call variant [8] |
   | --min-coverage-normal | Minimum coverage in normal to call somatic [8]           |
   | --min-var-freq        | Minimum variant frequency to call a heterozygote [0.10]  |
   | --somatic-p-value     | P-value threshold to call a somatic site [0.05]          |
   | --p-value             | P-value threshold to call a heterozygote [0.99]          |
   | --min-freq-for-hom    | Minimum frequency to call homozygote [0.75]              |
   | --strand-filter       | If set to 1, removes variants with >90% strand bias      |

   ```shell
   java -jar .../Varscan/VarScan.v2.4.3.jar somatic $pileup --output-snp $SNP --output-indel $Indel --mpileup 1 --min-coverage 20 --min-coverage-normal 20 --min-coverage-tumor 20 --min-var-freq 0.01 --somatic-p-value 0.05 --p-value 0.99 --min-freq-for-hom 0.75  --strand-filter 1 --output-vcf 1
   ```


2. ProcessSomatic

   ##### Isolating Calls by Type and Confidence

   The latest release of VarScan includes a new (undocumented) subcommand that will separate a somatic output file by somatic_status (Germline, Somatic, LOH). Somatic mutations will further be classified as high-confidence (.hc) or low-confidence (.lc). The command:

   ```
   java -jar VarScan.jar processSomatic [output.snp]
   ```

   ```shell
   java -jar.../Varscan/VarScan.v2.4.3.jar processSomatic --min-var-freq 0.01 ../Raw/$sample".SNP.vcf"
   java -jar .../Varscan/VarScan.v2.4.3.jar processSomatic --min-var-freq 0.01 ../Raw/$sample".Indel.vcf"
   ```

3. fpfilter

   ```shell
java -jar .../Varscan/VarScan.v2.4.3.jar fpfilter .../somatic.hc .../$expe".count" --output-file $sample"_Somatic_fil" --dream3-settings 1 
   ```





